### DISCUSSION TITLE
Why not use next as a devDependency?

### DISCUSSION THREAD

**Discussion by av-erencelik** [resend/react-email#1328] [Q&A] [answered]
I dont understand the usage of next as a normal dependency. It increases the image size at least 250 300 mb when i dockerized my node.js api.

**Comment by gabrielmfern** [comment, marked as answer]
Currently, we do not build the Next app before publishing, something in our to-do list is to have this in a better way.
I don't think it would be very hard to do otherwise either. 

You can see this from the `src/` directory on the `code` tab for the NPM package https://www.npmjs.com/package/react-email?activeTab=code.

The only thing to figure out how to run the Next app, while still avoiding having `next` as a dependency, is changing this:

https://github.com/resend/react-email/blob/20f560bb1cec1c7b4b97dcfcbaaccc0fbdc9e895/packages/react-email/src/cli/utils/preview/start-dev-server.ts#L123-L134

Having it not but also forces us into having the `@types` for certain packages as dependencies as well, which is a bit awkward I admit.

**Reply by gabrielmfern** [reply]
Would love some contributions on this!

**Reply by av-erencelik** [reply]
Thank you for your answer. It would probably make more sense to refactor the code as you mentioned. 

I solved my problem by adding the react-email package to my own project as devDependency. I continue to keep @react-email/components and @react-email/render packages as dependencies. As far as I can understand, the react-email package is required to use the development server. Please correct me if I'm wrong. 

In my use case, the react-email development server is not needed in the production deployment. This reduced my Docker image size from 950mb to 350mb.

If, as I mentioned, the react-email package is only required for the development server, I think it should be stated in the documentation that the react-email package can be added to the projects as devDependency, at least until you make the change you mentioned. I think it can at least help people who want to dockerize their application. 

As for contributions my hands are full now but when I have time I would love to make contributions especially about this topic!
